src/main/java/com/redhat/jenkins/plugins/cachet/CachetQueueTaskDispatcher.java

```diff
@@ -76,8 +76,8 @@ public CauseOfBlockage canRun(Queue.Item item) {
                                 msgs.add(message);
                                 log.info(message);
                                 requiredResourcesMetric.setGateUpdatedTime(System.currentTimeMillis());
-                                requiredResourcesMetric.setGatingStatus(r.getStatusName());
                             }
+                            requiredResourcesMetric.setGatingStatus(r.getStatusName());
```

Comment:
This will not work since once all resources are available, each one will get the status Operational and we will lose what it the blocking status was initially.
Please see #9

Reply:
:+1:

Reply:
@scoheb, So what is the blocking for?
If the resource is not Operational it will block the pipeline till it will be Operational again?
So it's not that different from the step I wrote.
Also it can block the step itself.

Reply:
@scoheb & @ggallen , please review, thanks :)